dbt/adapters/glue/gluedbapi/connection.py

```diff
@@ -382,7 +359,48 @@ def execute(cls,sql,output=True):
             for f in df.schema:
                 d[f.name] = record[f.name]
             results.append(({"type": "record", "data": d}))
-        dumped_results = json.dumps({"type": "results", "rowcount": rowcount,"results": results,"description": [{"name":f.name, "type":str(f.dataType)} for f in df.schema]},default=str)
+        description = [{"name":f.name, "type":str(f.dataType)} for f in df.schema]
+        raw_results = {"type": "results", "rowcount": rowcount, "results": results, "description": description}
+
+        if use_arrow:
+            s3_client = boto3.client('s3')
+            glue_client = boto3.client('glue')
+            security_config_name = glue_client.get_session(Id=session_id)["Session"].get("SecurityConfiguration", None)
+
+            o = urllib.parse.urlparse(location)
+            result_bucket = o.netloc
+            key = urllib.parse.unquote(o.path)[1:]
+            if not key.endswith('/'):
+                key = key + '/'
+            result_key = key + f"tmp-results/{session_id}.feather"
+
+            pdf = pd.DataFrame.from_dict(raw_results, orient="index")
+            feather.write_feather(pdf.transpose(), f"{session_id}.feather", "zstd")
+            
+            extra_args = {}
+            if security_config_name:
+                security_config = glue_client.get_security_configuration(Name=security_config_name)
+                s3_encryption = security_config["SecurityConfiguration"]["EncryptionConfiguration"].get("S3Encryption", None)
+                if s3_encryption and len(s3_encryption) > 0:
+                    s3_encryption_mode = s3_encryption[0]["S3EncryptionMode"]
+                    kms_key_arn = s3_encryption[0].get("KmsKeyArn", None)
+                    if s3_encryption_mode == "SSE-S3":
+                        extra_args["ServerSideEncryption"] = "AES256"
+                    elif s3_encryption_mode == "SSE-KMS":
+                        extra_args["ServerSideEncryption"] = "aws:kms"
+                        extra_args["SSEKMSKeyId"] = kms_key_arn.split("/")[1]
+            s3_client.upload_file(f"{session_id}.feather", result_bucket, result_key, ExtraArgs=extra_args)
+
+            # print and return only metadata instead of actual result data payload
+            del raw_results['results']
+            raw_results['result_bucket'] = result_bucket
+            raw_results['result_key'] = result_key
+            raw_results['description'] = description
+            dumped_results = json.dumps(raw_results, default=str)
+            print(dumped_results)
+            return dumped_results
```

Comment:
- Do we need these 3 lines?
- Do we need to call `print(dumped_results)` if `output` is false?

Reply:
Yes definitely.
The param use_arrow=True is always used with output=True, and stdout is used to pass those values to Interactive Sessions API, and this function needs to return here. (Returned value won't used actually but I kept it for consistency with other place)

Reply:
Thanks. Make sense. Would you comment it to the source code?

Reply:
Sure, added now.